### PR TITLE
Remove the Preview build-type

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This pull request removes the `preview` build-type, now that we no longer build `preview.va.gov`. It also does some additional cleanup of Vets.gov environments that I caught along the way.

Resolves https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15080

## Testing done
- Built the site
- Checked Jenkins dashboard

## Acceptance criteria
- [ ] No longer wasting resources on building preview

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/utilities/environment/isProduction.js [hunk 1]
```diff
@@ -1,10 +1,4 @@
 const prodEnvironments = [
-  // www.vets.gov
-  'production',
```
Comment (wyattwalter) [inline src/platform/utilities/environment/isProduction.js]: As far as our deployment tooling goes, www.vets.gov is still a production environment. I don't know what the impact of this particular change is, but just wanted to call this out and make sure it was purposeful.

Comment (ncksllvn) [reply src/platform/utilities/environment/isProduction.js]: Vets.gov builds from a separate entry point now (`script/vets-gov-to-va-gov/`), so any code that might work with this module won't ever build under the `production` environment. I'm not against leaving it though if we want that there for historical purposes.

File: va-gov/includes/header.html [hunk 2]
```diff
@@ -100,16 +100,6 @@
...
-{% if buildtype == 'preview' %}
-<div aria-live="polite" class="usa-alert usa-alert-info" style="margin: 0;">
-  <div class="usa-alert-body">
-    <h3 class="usa-alert-heading">You're viewing a preview of the new VA.gov</h3>
-    <p>We're building a new online experience to make it easier for Veterans to find, access, and manage the benefits they've earned. And we're building it with input from Veterans at every step. <b>Please note:</b> We work iteratively using agile development
-    practices, so you'll see new content, features, and functionality added to the site every day.</p>
-  </div>
-</div>
-{% endif %}
-
```
Comment (ncksllvn) [inline va-gov/includes/header.html]: Farewell preview banner 👋

File: src/platform/utilities/environment/isProduction.js (hunk 1 above)
Comment (wyattwalter) [reply src/platform/utilities/environment/isProduction.js]: Ah ok. I figured you were on top of it, just wanted to make sure. Carry on!

**Review by jbalboni — APPROVED** [review]

**Pull request merged**